File: Backend/app/src/main/java/com/example/demo/repository/CommentRepository.java
package com.example.demo.repository;

import com.example.demo.model.Comment;
import lombok.RequiredArgsConstructor;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;


@Repository
public interface CommentRepository extends JpaRepository<Comment, Long> {
    List<Comment> findByVideoIdOrderByCreatedAtDesc(Long videoId);

    Long countByVideoId(Long videoId);

}

File: Mobile/app/src/main/java/com/example/myapplication/data/repository/CommentRepository.kt
package com.example.myapplication.data.repository

import com.example.myapplication.data.api.RetrofitInstance
import com.example.myapplication.data.model.Comment


class CommentRepository {

    suspend fun getCommentsByVideoId(videoId: Long): List<Comment> {
        return RetrofitInstance.commentApi.getCommentsByVideoId(videoId)
    }

    suspend fun sendComment(comment: Comment) : Comment {
        return RetrofitInstance.commentApi.sendComment(comment)
    }

    suspend fun saveNewComment(content : String, videoId : Long) : Comment {
        return RetrofitInstance.commentApi.saveNewComment(content, videoId);
    }

    suspend fun countCommentsByVideoId(videoId: Long) : Long {
        return RetrofitInstance.commentApi.countCommentsByVideoId(videoId)
    }
}